provider/oracle/network/environ.go

```diff
@@ -152,8 +155,12 @@ func (e Environ) getSubnetInfo() ([]network.SubnetInfo, error) {
 
 // NetworkInterfaces is defined on the environs.Networking interface.
 func (e Environ) NetworkInterfaces(instId instance.Id) ([]network.InterfaceInfo, error) {
-	id := string(instId)
-	instance, err := e.client.InstanceDetails(id)
+	providerId, err := e.env.ProviderID(instId)
```

Comment:
Instances returns an ```instance.Instance``` array. I need to fetch the provider details for that particular instance, so I can generate the network interface details. One other option would be to cast the ```instance.Instance``` to ```OracleInstance```, but I would need to move some code around. I *think* the way it's written now, I might get into an import cycle if I try to import the provider code in the networking subpackage.

Another alternative is to expose the provider info for the machine, so we don't do 2 queries to the API. Thoughts?